<launch>
    <arg name="ee_tool"
        doc="Type of the tool attached to the end effector : [none, welding_laser, marker, logi_laser]"
        default="none" />
    <arg name="robot_model" default="ur10e" doc="Has to be one of the following : [ur5, ur10e]" />
    <arg name="robotName" value="$(arg robot_model)_robot" />
    <arg name="rosVersion" default="noetic" />

    <param name="ee_tool" value="$(arg ee_tool)" />
    <param name="robotName" value="$(arg robotName)" />
    <param name="rosVersion" value="$(arg rosVersion)" />

    <!-- use common launch file and pass all arguments to it -->
    <group if="$(eval arg('robot_model') == 'ur5')">
        <include file="$(find ur_description)/launch/load_ur5.launch" pass_all_args="true" />
    </group>

    <group if="$(eval arg('robot_model') == 'ur10e')">
        <include file="$(find ur_description)/launch/load_ur10e.launch" pass_all_args="true" />
    </group>

    <node name="test_robotic_arm" pkg="wp5_robotic_arms" type="test_robotic_arm"
        output="screen" >
        <param name="robotName" value="$(arg robotName)" />
        <param name="rosVersion" value="$(arg rosVersion)" />
    </node>
</launch>